File: app/src/main/java/br/com/nicetuber/util/JsonDeserializer.java
package br.com.nicetuber.util;

import com.google.gson.Gson;
import com.google.gson.JsonObject;

import br.com.nicetuber.model.ChannelStatistics;

/**
 * Created by thiagozg on 23/10/2017.
 */

public final class JsonDeserializer {

    public static ChannelStatistics deserializeChannelStatistics(Gson gson, JsonObject json) {
        JsonObject jsonObject = json.get("items").getAsJsonArray()
                                .get(0).getAsJsonObject()
                                .get("statistics").getAsJsonObject();
        return gson.fromJson(jsonObject, ChannelStatistics.class);
    }

}

File: app/src/main/java/br/com/nicetuber/util/JsonDeserializer.kt
package br.com.nicetuber.util

import br.com.nicetuber.model.ChannelStatistics
import com.google.gson.Gson
import com.google.gson.JsonObject

/**
 * Created by thiagozg on 23/10/2017.
 */

object JsonDeserializer {

    fun deserializeChannelStatistics(gson: Gson, json: JsonObject): ChannelStatistics {
        val jsonObject = json.get("items").asJsonArray
                .get(0).asJsonObject
                .get("statistics").asJsonObject
        return gson.fromJson(jsonObject, ChannelStatistics::class.java)
    }

}
